### ISSUE TITLE
Caused by: java.lang.ClassCastException: com.trello.lifecycle2.android.lifecycle.AndroidLifecycle_LifecycleAdapter cannot be cast to android.arch.lifecycle.GeneratedAdapter

### ISSUE THREAD

**Issue by gonemad** [trello/RxLifecycle#245]
I am getting this when i call bindToLifecycle now after updating to android arch lifecycle extensions 1.0.0-beta2 and android support library 26.1.0

Caused by: java.lang.ClassCastException: com.trello.lifecycle2.android.lifecycle.AndroidLifecycle_LifecycleAdapter cannot be cast to android.arch.lifecycle.GeneratedAdapter
                                                                  at android.arch.lifecycle.Lifecycling.createGeneratedAdapter(Lifecycling.java:79)
                                                                  at android.arch.lifecycle.Lifecycling.getCallback(Lifecycling.java:62)
                                                                  at android.arch.lifecycle.LifecycleRegistry$ObserverWithState.<init>(LifecycleRegistry.java:319)
                                                                  at android.arch.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:145)
                                                                  at com.trello.lifecycle2.android.lifecycle.AndroidLifecycle.<init>(AndroidLifecycle.java:36)
                                                                  at com.trello.lifecycle2.android.lifecycle.AndroidLifecycle.createLifecycleProvider(AndroidLifecycle.java:30)
                                                                  at com.trello.rxlifecycle2.android.lifecycle.kotlin.RxlifecycleKt.bindToLifecycle(rxlifecycle.kt:23)

Looking at the code in the lifecycle arch lib.. i believe the AndroidLifecycle class needs to inherit from GenericLifecycleObserver or FullLifecycleObserver now

**Comment by dlew** [comment]
We'll have to upgrade the code to match the betas. I think this is probably alright because the beta APIs shouldn't be changing dramatically anymore.

**Comment by gonemad** [comment]
Yup should be a stable api now

Release Candidate - October 18, 2017
All major artifacts (except Paging) are now 1.0.0-rc1.

**Comment by micHar** [comment]
Same here, we have stable lifecycle for quite some time now, do you plan a release to match it soon?

**Comment by dlew** [comment]
My life is fairly hectic at the moment, but I'll see if I can find time in the next week to do another release.

**Comment by a1018875550** [comment]
have same error too;

**Comment by bagrusss** [comment]
I have the same problem
```
dependencies {

    ...

    implementation "android.arch.lifecycle:runtime:1.0.3"
    implementation "android.arch.lifecycle:extensions:1.0.0"
    annotationProcessor "android.arch.lifecycle:compiler:1.0.0"

    implementation 'com.trello.rxlifecycle2:rxlifecycle-android-lifecycle-kotlin:2.2.0'

    ...
}
```

Have you any progress for this problem?

**Comment by dlew** [comment]
Life continues to be hectic and I apologize for that. I will really, really try to find time to get to this this week.

**Comment by a1018875550** [comment]
when I remove the paging library ```implementation "android.arch.paging:runtime:1.0.0-alpha3"``` ,it's fine

**Comment by gonemad** [comment]
I stopped using this lib but for what its worth when i filed this issue i was not using the paging library and my investigation into the issue seemed to indicate nothing related to the paging library. I would not close this without verifying that

**Comment by dlew** [comment]
It was closed because #249 was merged, which fixes the issue.

**Comment by gonemad** [comment]
Nm ignore that. Didnt see it was fixed
